pilot/pkg/model/sidecar.go

```diff
@@ -845,34 +845,29 @@ func serviceMatchingVirtualServicePorts(service *Service, vsDestPorts sets.Set[i
 }
 
 // computeWildcardHostVirtualServiceIndex computes the wildcardHostVirtualServiceIndex for a given
-// list of virtualServices. This is used to optimize the lookup of the most specific wildcard host
+// (sorted) list of virtualServices. This is used to optimize the lookup of the most specific wildcard host.
+//
+// N.B the caller MUST presort virtualServices based on the desired precedence for duplicate hostnames.
+// This function will persist that order and not overwrite any previous entries for a given hostname.
 func computeWildcardHostVirtualServiceIndex(virtualServices []config.Config, services []*Service) map[host.Name]types.NamespacedName {
 	fqdnVirtualServiceHostIndex := make(map[host.Name]config.Config, len(virtualServices))
 	wildcardVirtualServiceHostIndex := make(map[host.Name]config.Config, len(virtualServices))
 	for _, vs := range virtualServices {
 		v := vs.Spec.(*networking.VirtualService)
 		for _, h := range v.Hosts {
-			// We may have duplicate (not just overlapping) hosts; in the case of exact duplicates, take the oldest one first
+			// We may have duplicate (not just overlapping) hosts; assume the list of VS is sorted already
+			// and never overwrite existing entries
 			if host.Name(h).IsWildCarded() {
-				existingVs, exists := wildcardVirtualServiceHostIndex[host.Name(h)]
+				_, exists := wildcardVirtualServiceHostIndex[host.Name(h)]
 				if !exists {
 					wildcardVirtualServiceHostIndex[host.Name(h)] = vs
-					continue
```

Comment:
why remove the continue

Reply:
I think this is to reduce unnecessary loop

Reply:
Because it's an if-else, the continue doesn't currently do anything. The continue on 869 could be removed too I think. I'm also on my phone so I may be misreading something though

Reply:
you are right L869 continue need to be removed

Reply:
Removed